AutoCollection/StatsBeat.ts

```diff
@@ -191,7 +198,9 @@ export class StatsBeat {
                 "feature": this._features,
             }
         };
-        this._client.trackMetric(statsBeat);
+        let envelope = EnvelopeFactory.createEnvelope(statsBeat, Contracts.TelemetryType.Metric, null, null, this._statsBeatConfig);
+        envelope.name = Constants.StatsBeatMetricName;
+        this._sender.send([envelope]);
```

Comment:
As send is async (returning a promise) we should also handle this as async -- otherwise if the promise rejects it will cause a promise unhandledrejection to be thrown / caught by something -- not sure if this would be terminal for node environment

Reply:
This code must be handling all exceptions already so in theory it must not throw any error making the promise to be rejected, even so I added async and try and catch to be sure nothing goes wrong